### ISSUE TITLE
[Feature Request] Random Looping Option

### ISSUE THREAD

**Issue by GravyPouch** [henriquehbr/svelte-typewriter#23]
Would it be possible to add a random option to the loop function. As I would like to the text that loops to randomly pick a different line of text to print. I will try to implement this myself but I will have to learn a few things first.

**Comment by henriquehbr** [comment]
Hi there @GravyPouch!

Thanks for the suggestion, actually, i think it would be fairly easy to implement such behavior on the loop mode, but i'm not sure which would be the best way (in terms of clean code) of passing this `random` parameter to the `loop` function, here are some of the solutions i could think of:

- Passing a "options" object as second parameter (or even, the first and only one) of the `loop` prop, having as properties, the `interval` between each word, and `random`, which would define if the words should be randomly picked, in terms of code, it would look like this:

```svelte
<Typewriter loop={{ interval: 2000, random: true }}>
    <p>Lorem</p>
    <p>Ipsum</p>
    <p>Dolor</p>
</Typewriter>
```

> To be honest, quite don't like this solution, because of the need of having to pass a object as parameter to the prop, making it less readable than if it was just a `number` or `boolean`

- Creating a new animation mode `loopRandom`, which will share most of it's internal logic with the already existing `loop` mode, this new prop would require only a single parameter (just like [`loop`](https://github.com/henriquehbr/svelte-typewriter#modes) mode at the moment), here's a preview:

```svelte
<Typewriter loopRandom={2000}>
    <p>Lorem</p>
    <p>Ipsum</p>
    <p>Dolor</p>
</Typewriter>
```

Any suggestions would be greatly appreciated :)

**Comment by GravyPouch** [comment]
That sounds great. However you would like to do I am sure it will work well. I agree with you that I like the simplicity of loopRandom, and only having to pass a number.

**Comment by henriquehbr** [comment]
Alright, let's stick with the `loopRandom` alternative, i'll start working on it as soon as possible (as for now i'm finishing some pending stuff on other side-projects)

Thank you for the valuable feedback!

**Comment by henriquehbr** [comment]
> This feature was implemented in #24 and published to npm on version `v2.4.1` (i strongly recommend this version due to a already fixed bug related to single text node elements that was accidentally released on `v2.4.0`)

About the random loop behavior, during the development i had to choose between two different behaviors to the new `loopRandom` mode:

1. Make the loop go through each word, animate it, and store in a array to avoid frequently repeated words, when there are no more words to animate, the array is emptied and the process is restarted
2. Make the loop go through each word, animate it, and only store the previous word, making the effect a bit more random in terms of word picking, but also slightly more susceptible to repeated words

i manually tested it with a bunch of possible scenarios (nested elements, parent elements with single text node, etc) and everything seems to be working fine, hope it serves you well 😄 

Thanks for your contribution, if you experience any issue or unexpected behavior with `svelte-typewriter` or this feature in particular, you can report it here and i will try to solve as soon as possible
